dockerfile

```diff
@@ -1,37 +1,35 @@
-# Dockerfile
-## Build venv
-FROM python:3.10.5-buster
-
-RUN mkdir /app 
-COPY /app /app
-COPY pyproject.toml /app 
-WORKDIR /app
-ENV PYTHONPATH=${PYTHONPATH}:${PWD} 
-RUN pip3 install poetry
-RUN poetry config virtualenvs.create false
-RUN poetry install --no-dev
-
-
-
-
-
-
-
-# Install poetry, see https://python-poetry.org/docs/#installation
-ENV POETRY_VERSION=1.1.1
-RUN curl -sSL https://raw.githubusercontent.com/python-poetry/poetry/master/get-poetry.py | python
-ENV PATH /root/.poetry/bin:$PATH
-
-# Install dependencies
-WORKDIR /app
-RUN python -m venv /app/venv
-COPY pyproject.toml poetry.lock ./
-RUN . /app/venv/bin/activate && poetry install
-ENV PATH /app/venv/bin:$PATH
-
-# Add code
-COPY . ./
-
-#HEALTHCHECK --start-period=30s CMD python -c "import requests; requests.get('http://localhost:8000', timeout=2)"
-
-CMD ["gunicorn", "blog.wsgi", "-b 0.0.0.0:8000", "--log-file", "-"]
\ No newline at end of file
+FROM python:3.8.6-buster as base
+RUN apt-get update
+
+WORKDIR /opt
+COPY . /opt
+RUN pip install poetry && poetry install
```

Comment:
You could optimise this.

Currently, if you modify any file at all, the next docker build has to run from line 5. So every time you update your image, you are reinstalling Poetry and reinstalling all your dependencies.

Instead, consider this:
```
RUN pip install poetry
COPY pyproject.toml poetry.lock /opt/
RUN poetry install
COPY ./todo_app /opt/todo_app/
```

Now, you never have to reinstall poetry again. Whenever you rebuild your image it will take the cached layer that has Poetry installed and go from there.

In addition, if you just update files in the todo_app folder (without modifying pyproject.toml), then it can also skip reinstalling dependencies. It will run from the COPY todo_app line.